snapcraft/plugins/gradle.py

```diff
@@ -110,3 +119,27 @@ def _get_proxy_options(self):
                     proxy_options.append(
                         '-D{}.proxyPort={}'.format(var, parsed_url.port))
         return proxy_options
+
+
+def _link_or_copy(source, destination, boundary):
```

Comment:
Can I recommend this do a separate PR?
I copied this from the `dump` plugin.  Where I should really just be using file_utils.link_or_copy, which is what I'm going to check in.
https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/dump.py